comment/models/flags.py

```diff
@@ -87,7 +87,7 @@ def toggle_flagged_state(self):
         if not allowed_flags:
             return
         self.refresh_from_db()
-        if self.count > allowed_flags and self.state == self.UNFLAGGED:
+        if self.count > allowed_flags and self.state not in [self.RESOLVED, self.REJECTED]:
```

Comment:
something that has a state rejected shall never be flagged in future now !!!

The earlier version seemed more appropriate to me.

Reply:
This is how it should be, when the flag state is resolved or rejected, that means the moderator has viewd the comment and took an action, thus the flag state won't be set to flagged anymore.